packages/nimbus/src/components/data-table/components/data-table.header.tsx

```diff
@@ -0,0 +1,212 @@
+            margin: "-1px", 
+            overflow: "hidden", 
+            clip: "rect(0, 0, 0, 0)", 
+            whiteSpace: "nowrap", 
+            border: 0 
+          }}>
+            Expand rows
+          </span>
+        </RaColumn>
+      )}
+
+      {visibleCols.map((col, index) => {
+        const isSortable = allowsSorting && col.isSortable !== false;
+        const isLastColumn = index === visibleCols.length - 1;
+        return (
+          <>
+            <RaColumn
+              allowsSorting={isSortable}
+              key={col.id}
+              id={col.id}
+              isRowHeader
+              width={col.width}
+              defaultWidth={col.defaultWidth}
+              minWidth={col.minWidth}
+              maxWidth={col.maxWidth}
+            >
+              {isHeaderHovered && !isLastColumn && (
+                <Divider
+                  orientation="vertical"
+                  color="gray.200"
+                  className="data-table-header-divider"
+                />
+              )}
+              <div
+                tabIndex={0}
+                onFocus={() => setFocusedColumn(col.id)}
+                onMouseEnter={() => setHoveredColumn(col.id)}
+                onMouseLeave={() => setHoveredColumn(null)}
+                style={{
+                  cursor: isSortable ? "pointer" : "default",
+                }}
+              >
+                <span data-multiline-header>{col.header}</span>
+                {col.headerIcon && (
+                  <span style={{ marginLeft: "8px" }}>{col.headerIcon}</span>
+                )}
+                {renderSortIndicator(col.id)}
+              </div>
+              {col.isAdjustable !== false && !isLastColumn && (
+                <ColumnResizer aria-label="Resize column">
+                  {({ isResizing, isFocused }) => (
+                    <DataTableColumnResizer
+                      data-resizing={isResizing}
+                      data-focused={isFocused}
+                    />
+                  )}
+                </ColumnResizer>
+              )}
+            </RaColumn>
+
+            {/* Details column header - shown after first data column */}
+            {showDetailsColumn && index === 0 && (
+              <RaColumn
+                key="details-header"
+                id="details"
+                className="details-column-header"
+                width={70}
+                allowsSorting={false}
+                aria-label="Row details"
+              >
+                <span style={{ 
+                  position: "absolute", 
+                  width: "1px", 
+                  height: "1px", 
+                  padding: 0, 
+                  margin: "-1px", 
+                  overflow: "hidden", 
+                  clip: "rect(0, 0, 0, 0)", 
+                  whiteSpace: "nowrap", 
+                  border: 0 
+                }}>
+                  Details
+                </span>
+              </RaColumn>
```

Comment:
I'm not sure I like the 'show details' button in a column that always renders after the first column of data. I know of at least one table (audit log) where this is not the case. I personally think this should be added by consumers in whichever cell makes sense to them. In any case, I definitely think this needs more discussion.